planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -161,26 +161,35 @@ class MoveGroup::MoveGroupImpl
     ros::Time start_time = ros::Time::now();
     while (start_time == ros::Time::now())
     {
-      ros::WallDuration(0.01).sleep();
+      ros::WallDuration(0.001).sleep();
       ros::spinOnce();
+      if( node_handle_.getCallbackQueue() ) {
+        ( ( ros::CallbackQueue * ) node_handle_.getCallbackQueue())->callAvailable();
+      }
     }
 
     // wait for the server (and spin as needed)
     if (wait_for_server == ros::Duration(0, 0))
     {
       while (node_handle_.ok() && !action->isServerConnected())
       {
-        ros::WallDuration(0.02).sleep();
+        ros::WallDuration(0.001).sleep();
```

Comment:
It really makes a difference, the connection happens faster. 0.001 is already quite a long sleep in computer terms for me.